src/main/java/net/snowflake/ingest/streaming/internal/BlobBuilder.java

```diff
@@ -89,13 +90,21 @@ static <T> Blob constructBlobAndMetadata(
         final byte[] compressedChunkData;
         final int chunkLength;
         final int compressedChunkDataSize;
+        int extendedMetadataSize = -1;
 
         if (internalParameterProvider.getEnableChunkEncryption()) {
           Pair<byte[], Integer> paddedChunk =
               padChunk(serializedChunk.chunkData, Constants.ENCRYPTION_ALGORITHM_BLOCK_SIZE_BYTES);
           byte[] paddedChunkData = paddedChunk.getFirst();
           chunkLength = paddedChunk.getSecond();
 
+          if (internalParameterProvider.getComputeExtendedMetadataSize()) {
+            extendedMetadataSize =
```

Comment:
this is risky in that (a) the behavior of paddedChunkData can change and break the assumption here of no-padding-at-the-tail, (b) no guaranteed way of avoiding a corrupt value from flowing out.

1. lets extract extended metadata size before padding. 
2. verify that the last four bytes are == ParquetFileWriter.MAGIC
3. Is there nothing on ParquetFileWriter that exposes the extendedMetadataSize?

Reply:
1. The `chunkLength` returned by `padChunk` is the length before padding, the offset should be correct.
2. Added check in `Utils.getExtendedMetadataSize`.
3. Afaik the parquet file writer only log this value ([ref](https://github.com/apache/parquet-java/blob/eed26b4ea96881cc543b4233cdddece5d1d608d7/parquet-hadoop/src/main/java/org/apache/parquet/hadoop/ParquetFileWriter.java#L1918))